x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/UnsupportedAttribute.java

```diff
@@ -70,23 +73,35 @@ public UnsupportedAttribute(Source source, String name, UnsupportedEsField field
         this.message = customMessage == null ? errorMessage(qualifiedName(), field) : customMessage;
     }
 
-    public UnsupportedAttribute(StreamInput in) throws IOException {
+    private UnsupportedAttribute(StreamInput in) throws IOException {
         this(
             Source.readFrom((PlanStreamInput) in),
             in.readString(),
-            new UnsupportedEsField(in),
+            in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ATTRIBUTE_CACHED_SERIALIZATION_8_15)
```

Comment:
I know this is our current convention, but I think it's somewhat of a lie.

I think it would be more correct if `main` received another, even newer transport version in https://github.com/elastic/elasticsearch/pull/112866, `THE_NEWEST_VERSION`, and this read:
```
isPatchFrom(ESQL_ATTRIBUTE_CACHED_SERIALIZATION_8_15) || onOrAfter(THE_NEWEST_VERSION)
```
This would mean  we'd have to ad-hoc define `THE_NEWEST_VERSION` also in here, without adding it to `TransportVersions.java` to avoid broadcasting the wrong transport version. And hoping that no reflections pick up on this transport version definition or so.

OTOH, since 8.16 is not yet released and communication between 8.15 and serverless nodes is irrelevant (I _think_), this is probably fine as-is, because we only break transport between 8.15 and unreleased commits before 8.16's release.

Reply:
I don't think a new version bump in main will bring practical advantages, we would still have to check the old ESQL_ATTRIBUTE_CACHED_SERIALIZATION for Serverless compatibility